lib/src/inline_parser.dart

```diff
@@ -346,38 +345,38 @@ abstract class InlineSyntax {
 
   /// The first character of [pattern], to be used as an efficient first check
   /// that this syntax matches the current parser position.
-  final int _startCharacter;
+  final int? _startCharacter;
 
   /// Create a new [InlineSyntax] which matches text on [pattern].
   ///
   /// If [startCharacter] is passed, it is used as a pre-matching check which
   /// is faster than matching against [pattern].
-  InlineSyntax(String pattern, {int startCharacter})
+  InlineSyntax(String pattern, {int? startCharacter})
       : pattern = RegExp(pattern, multiLine: true),
         _startCharacter = startCharacter;
 
   /// Tries to match at the parser's current position.
   ///
   /// The parser's position can be overriden with [startMatchPos].
   /// Returns whether or not the pattern successfully matched.
-  bool tryMatch(InlineParser parser, [int startMatchPos]) {
+  bool tryMatch(InlineParser parser, [int? startMatchPos]) {
     startMatchPos ??= parser.pos;
 
     // Before matching with the regular expression [pattern], which can be
     // expensive on some platforms, check if even the first character matches
     // this syntax.
     if (_startCharacter != null &&
-        parser.source.codeUnitAt(startMatchPos) != _startCharacter) {
+        parser.source!.codeUnitAt(startMatchPos) != _startCharacter) {
       return false;
     }
 
-    final startMatch = pattern.matchAsPrefix(parser.source, startMatchPos);
+    final startMatch = pattern.matchAsPrefix(parser.source!, startMatchPos);
     if (startMatch == null) return false;
 
     // Write any existing plain text up to this point.
     parser.writeText();
 
-    if (onMatch(parser, startMatch)) parser.consume(startMatch[0].length);
+    if (onMatch(parser, startMatch)) parser.consume(startMatch.match.length);
```

Comment:
where does `match` come from?

Reply:
It's a secret!

No, its an extension I dropped into `lib/util.dart`. I think Lasse said it's being considered for the Match class, because `match[0]` should always be non-nullable, but `match[1]` has to be nullable. Lots of `[1]!` and `[2]!` sprinkled :(